common/src/main/java/com/wynntils/models/containers/BankModel.java

```diff
@@ -83,48 +96,95 @@ public void onScreenClose(ScreenClosedEvent e) {
     }
 
     public Optional<String> getPageName(int page) {
-        Storage<Map<Integer, String>> pageNamesMap = getCurrentNameMap();
+        Map<Integer, String> pageNamesMap = getCurrentNameMap();
 
         if (pageNamesMap == null) return Optional.empty();
 
-        return Optional.ofNullable(pageNamesMap.get().get(page));
+        return Optional.ofNullable(pageNamesMap.get(page));
     }
 
     public void saveCurrentPageName(String nameToSet) {
-        Storage<Map<Integer, String>> pageNamesMap = getCurrentNameMap();
+        switch (currentContainer) {
+            case ACCOUNT_BANK -> {
+                customBankPageNames.get().put(currentPage, nameToSet);
+                customBankPageNames.touched();
+            }
+            case BLOCK_BANK -> {
+                customBlockBankPageNames.get().put(currentPage, nameToSet);
+                customBlockBankPageNames.touched();
+            }
+            case BOOKSHELF -> {
+                customBookshelfPageNames.get().put(currentPage, nameToSet);
+                customBookshelfPageNames.touched();
+            }
+            case CHARACTER_BANK -> {
+                Map<Integer, String> nameMap = new TreeMap<>();
 
-        if (pageNamesMap == null) return;
+                if (customCharacterBankPagesNames.get().containsKey(Models.Character.getId())) {
+                    nameMap = customCharacterBankPagesNames.get().get(Models.Character.getId());
+                }
 
-        pageNamesMap.get().put(currentPage, nameToSet);
-        pageNamesMap.touched();
+                nameMap.put(currentPage, nameToSet);
+
+                customCharacterBankPagesNames.get().put(Models.Character.getId(), nameMap);
+                customCharacterBankPagesNames.touched();
+            }
+            case MISC_BUCKET -> {
+                customMiscBucketPageNames.get().put(currentPage, nameToSet);
+                customMiscBucketPageNames.touched();
+            }
+        }
 
         editingName = false;
     }
 
     public void resetCurrentPageName() {
-        Storage<Map<Integer, String>> pageNamesMap = getCurrentNameMap();
-
-        if (pageNamesMap == null) return;
-
-        pageNamesMap.get().remove(currentPage);
-        pageNamesMap.touched();
+        switch (currentContainer) {
+            case ACCOUNT_BANK -> {
+                customBankPageNames.get().remove(currentPage);
+                customBankPageNames.touched();
+            }
+            case BLOCK_BANK -> {
+                customBlockBankPageNames.get().remove(currentPage);
+                customBlockBankPageNames.touched();
+            }
+            case BOOKSHELF -> {
+                customBookshelfPageNames.get().remove(currentPage);
+                customBookshelfPageNames.touched();
+            }
+            case CHARACTER_BANK -> {
+                customCharacterBankPagesNames.get().putIfAbsent(Models.Character.getId(), new TreeMap<>());
```

Comment:
Seemingly you put it here, where a getOrDefault could work below.

Reply:
What would the default be?

Reply:
An empty map